lib/optimize/CommonsChunkPlugin.js

```diff
@@ -2,183 +2,189 @@
+								} else {
+									commonModulesCount[idx]++;
+								}
+							});
+						});
+						const _minChunks = (minChunks || Math.max(2, usedChunks.length))
+						commonModulesCount.forEach((count, idx) => {
+							const module = commonModules[idx];
+							if(typeof minChunks === "function") {
+								if(!minChunks(module, count))
+									return;
+							} else if(count < _minChunks) {
+								return;
+							}
+							if(module.chunkCondition && !module.chunkCondition(commonChunk))
+								return;
+							reallyUsedModules.push(module);
+						});
+					}
+					if(minSize) {
+						const size = reallyUsedModules.reduce((a, b) => {
+							return a + b.size();
+						}, 0);
+						if(size < minSize)
 							return;
-						chunk.blocks.forEach(function(block) {
-							block.chunks.unshift(commonChunk);
-							commonChunk.addBlock(block);
+					}
+					const reallyUsedChunks = [];
+					reallyUsedModules.forEach((module) => {
+						usedChunks.forEach((chunk) => {
+							if(module.removeChunk(chunk)) {
+								if(reallyUsedChunks.indexOf(chunk) < 0)
+									reallyUsedChunks.push(chunk);
+							}
 						});
+						commonChunk.addModule(module);
+						module.addChunk(commonChunk);
 					});
-					asyncChunk.origins = reallyUsedChunks.map(function(chunk) {
-						return chunk.origins.map(function(origin) {
-							var newOrigin = Object.create(origin);
-							newOrigin.reasons = (origin.reasons || []).slice();
-							newOrigin.reasons.push("async commons");
-							return newOrigin;
+					if(asyncOption) {
+						reallyUsedChunks.forEach((chunk) => {
+							if(chunk.isInitial())
+								return;
+							chunk.blocks.forEach((block) => {
+								block.chunks.unshift(commonChunk);
+								commonChunk.addBlock(block);
+							});
 						});
-					}).reduce(function(arr, a) {
-						arr.push.apply(arr, a);
-						return arr;
-					}, []);
-				} else {
-					usedChunks.forEach(function(chunk) {
-						chunk.parents = [commonChunk];
-						chunk.entrypoints.forEach(function(ep) {
-							ep.insertChunk(commonChunk, chunk);
+						asyncChunk.origins = reallyUsedChunks.map((chunk) => {
+							return chunk.origins.map((origin) => {
+								const newOrigin = Object.create(origin);
+								newOrigin.reasons = (origin.reasons || []).slice();
+								newOrigin.reasons.push("async commons");
+								return newOrigin;
+							});
+						}).reduce((arr, a) => {
+							arr.push.apply(arr, a);
+							return arr;
+						}, []);
+					} else {
+						usedChunks.forEach((chunk) => {
+							chunk.parents = [commonChunk];
+							chunk.entrypoints.forEach((ep) => {
+								ep.insertChunk(commonChunk, chunk);
+							});
+							commonChunk.addChunk(chunk);
 						});
-						commonChunk.addChunk(chunk);
-					});
-				}
-				if(filenameTemplate)
-					commonChunk.filenameTemplate = filenameTemplate;
-			}, this);
-			return true;
+					}
+					if(filenameTemplate)
+						commonChunk.filenameTemplate = filenameTemplate;
+				}, this);
```

Comment:
`processCommonChunk` could be an arrow function instead; it doesn't use `this`, and it's being bound with `this` by the second argument to `forEach` anyway. But that would lose the `Function#name` in stack traces 😕